packages/@aws-cdk/core/lib/lazy.ts

```diff
@@ -100,28 +140,187 @@ export interface LazyAnyValueOptions {
  * will only be calculated later, during synthesis.
  */
 export class Lazy {
+  /**
+   * Defer the calculation of a string value to synthesis time
+   *
+   * Use this if you want to render a string to a template whose actual value depends on
+   * some state mutation that may happen after the construct has been created.
+   *
+   * If you are simply looking to force a value to a `string` type and don't need
+   * the calculation to be deferred, use `Token.asString()` instead.
+   *
+   * @deprecated Use `Lazy.string()` or `Lazy.uncachedString()` instead.
+   */
   public static stringValue(producer: IStringProducer, options: LazyStringValueOptions = {}) {
-    return Token.asString(new LazyString(producer), options);
+    return Token.asString(new LazyString(producer, false), options);
+  }
+
+  /**
+   * Defer the one-time calculation of a string value to synthesis time
+   *
+   * Use this if you want to render a string to a template whose actual value depends on
+   * some state mutation that may happen after the construct has been created.
+   *
+   * If you are simply looking to force a value to a `string` type and don't need
+   * the calculation to be deferred, use `Token.asString()` instead.
+   *
+   * The inner function will only be invoked one time and cannot depend on
+   * resolution context.
+   */
+  public static string(producer: IStableStringProducer, options: LazyStringValueOptions = {}) {
+    return Token.asString(new LazyString(producer, true), options);
   }
 
+  /**
+   * Defer the calculation of a string value to synthesis time
+   *
+   * Use if this function is not recommended; unless you know you need it for sure, you
+   * probably don't. Use `Lazy.string()` instead.
+   *
+   * The inner function may be invoked multiple times during synthesis. You should
+   * only use this function if the calculated value may change during the 'prepare'
+   * phase of synthesis.
```

Comment:
Isn't the prepare phase going away at v2? 

Nevertheless, it would be nice for a better, more customer friendly explanation. Think of a customer coming across this for the first time.